tests/unit/resource/cross_pool_injection.cpp

```diff
@@ -0,0 +1,258 @@
+    {
+        // normal priority
+        std::size_t random_pool_1 = st_rand() % num_pools;
+        std::size_t random_pool_2 = st_rand() % num_pools;
+        auto& exec_3 = NP_executors[random_pool_1];
+        auto& exec_4 = NP_executors[random_pool_2];
+        auto f3 = hpx::async(exec_3, &dummy_task, 0);
+        auto f4 = f3.then(exec_4, [=, &counter](auto&&) {
+            dummy_task(0);
+            --counter;
+        });
+    }
+    do
+    {
+        hpx::this_thread::yield();
+    } while (counter > 0);
+
+    std::cout << "3: Starting HP->NP " << loops << std::endl;
+    counter = loops;
+    for (int i = 0; i < loops; ++i)
+    {
+        // mixed priority, HP->NP
+        std::size_t random_pool_1 = st_rand() % num_pools;
+        std::size_t random_pool_2 = st_rand() % num_pools;
+        auto& exec_5 = HP_executors[random_pool_1];
+        auto& exec_6 = NP_executors[random_pool_2];
+        auto f5 = hpx::async(exec_5, &dummy_task, 0);
+        auto f6 = f5.then(exec_6, [=, &counter](auto&&) {
+            dummy_task(0);
+            --counter;
+        });
+    }
+    do
+    {
+        hpx::this_thread::yield();
+    } while (counter > 0);
+
+    std::cout << "4: Starting NP->HP " << loops << std::endl;
+    counter = loops;
+    for (int i = 0; i < loops; ++i)
+    {
+        // mixed priority, NP->HP
+        std::size_t random_pool_1 = st_rand() % num_pools;
+        std::size_t random_pool_2 = st_rand() % num_pools;
+        auto& exec_7 = NP_executors[random_pool_1];
+        auto& exec_8 = HP_executors[random_pool_2];
+        auto f7 = hpx::async(exec_7, &dummy_task, 0);
+        auto f8 = f7.then(exec_8, [=, &counter](auto&&) {
+            dummy_task(0);
+            --counter;
+        });
+    }
+    do
+    {
+        hpx::this_thread::yield();
+    } while (counter > 0);
+
+    std::cout << "5: Starting suspending " << loops << std::endl;
+    counter = loops;
+    for (int i = 0; i < loops; ++i)
+    {
+        // tasks that depend on each other and need to suspend
+        std::size_t random_pool_1 = st_rand() % num_pools;
+        std::size_t random_pool_2 = st_rand() % num_pools;
+        auto& exec_7 = NP_executors[random_pool_1];
+        auto& exec_8 = HP_executors[random_pool_2];
+        // random delay up to 5 miliseconds
+        std::size_t delay = st_rand() % 5;
+        auto f7 = hpx::async(exec_7, &dummy_task, delay);
+        auto f8 = hpx::async(exec_8, [f7(std::move(f7)), &counter]() mutable {
+            // if f7 is not ready then f8 will suspend itself on get
+            f7.get();
+            dummy_task(0);
+            --counter;
+        });
+    }
+    do
+    {
+        hpx::this_thread::yield();
+    } while (counter > 0);
+
+    return hpx::finalize();
+}
+
+int main(int argc, char* argv[])
+{
+    std::size_t max_threads = std::thread::hardware_concurrency();
+    std::vector<std::string> cfg = {
+        "hpx.os_threads=" + std::to_string(max_threads)};
```

Comment:
This should not be needed if we set the number of threads for the test. Or this should be `hpx.os_threads=all`.

Reply:
ok. I fix that/. probably just a left over cut'n'paste from another test